internal/ethereum/blocklistener.go

```diff
@@ -40,6 +40,12 @@ type blockUpdateConsumer struct {
 	updates chan<- *ffcapi.BlockHashEvent
 }
 
+type blockFilterStatus struct {
+	isEstablished bool
+	signal        chan struct{}
+	mux           sync.Mutex
```

Comment:
don't think this mutex is actually required as the upstream logic already ensures there is a single thread modify the channel, so a flag + a channel should does the job.

Can we name them consistently with existing methods name. e.g. `isStarted` & `startDone`?